client/src/App.js

```diff
@@ -6,18 +6,19 @@ import Footer from "./Components/Footer/Footer";
 import Profile from "./Pages/UserProfile";
 import profileData from "./Data/profileData.json";
 import "./styles.css";
+import NotFoundPage from './pages/NotFoundPage';
 
 function App() {
   return (
     <BrowserRouter>
       <Header />
       <Switch>
         <Route path="/profile" component={Profile} />
```

Comment:
Make it as a persisted route. Just change this to `<Route path="/profile" component={Profile} exact />`